### PR TITLE
Galaxy meta docs table

### PR DESCRIPTION
**Pull request by abadger**
##### SUMMARY
Update the generated collections meta doc with an rst table

##### ISSUE TYPE

- Docs Pull Request

##### ADDITIONAL INFORMATION
This is built on top of https://github.com/ansible/ansible/pull/60148 and that should be merged first and this second.

### CODE REVIEW

File: lib/ansible/galaxy/data/collections_galaxy_meta.yml [hunk 1]
```diff
@@ -7,11 +7,10 @@
 # type: The type of value that can be set, aligns to the values in the plugin formatter
 ---
 - key: namespace
-  description:
-  - The namespace of the collection.
-  - This can be a company/brand/organization or product namespace under which all content lives.
-  - May only contain alphanumeric characters and underscores. Additionally namespaces cannot start with underscores or
-    numbers and cannot contain consecutive underscores.
+  description: |
```
Comment (jborean93) [inline lib/ansible/galaxy/data/collections_galaxy_meta.yml]: Why this change, it seems to move away from how we handle entries in module docs where a new list entry in a newline and now we can't wrap the text at 120 characters without it being a literal newline in the description.

Comment (abadger) [reply lib/ansible/galaxy/data/collections_galaxy_meta.yml]: > Also change the structure of one description to test that both ways of
formatting the description work

I was debating whether to change this back after testing it.... On the one hand, changing it back will make everything match.  On the other hand, there would be nothing testing that both ways work so it will be very easy for this formatting to break.

Perhaps it would be better if we remove the code supporting the second form of formatting instead?

(Also note, it's quite possible that module docs work with both syntaxes as well but we just don't have docs which are using that either...)

Comment (abadger) [reply lib/ansible/galaxy/data/collections_galaxy_meta.yml]: I just tested what module docs do.  The module description actually renders a list as a list.  It tries to render double-newline spaced paragraphs but is buggy (It renders paragraphs but starts with an initial bullet so that there is an error in the output)

The parameter descriptions must be a list.  Newlines are swallowed rather than rendering new paragraphs.

Comment (abadger) [reply lib/ansible/galaxy/data/collections_galaxy_meta.yml]: After jborean prompted me to look at this further, the code in the template isn't there for the reason that I thought.  The API between reading the source material and passing to the template could be changed to make it clearer what's happening.  I'll fix that up both here and in the plugin_docs.

Comment (abadger) [reply lib/ansible/galaxy/data/collections_galaxy_meta.yml]: Fixed now.

plugin_docs actually already normalized options['description'] (but not the main module description) in the plugin portion.  The template just needed to have the logic stripped out of it.  (I updated the module description to match).

collection_meta now does the same thing: normalizes the data in the plugin and then the template can do its work without two code paths.

**Comment by abadger** [comment]
Here's a screenshot of the new table:

https://toshio.fedorapeople.org/ansible/new-collection-meta.png

It's pretty close to the present table so I think it's good to go.

**Review by jborean93 — APPROVED** [review]

**Comment by acozine** [comment]
Thanks @abadger and @jborean93.
